standalone/matRad_compileStandalone.m

```diff
@@ -0,0 +1,55 @@
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% Script to compile an executable matRad version with the matlab
+% application compiler.
+%
+% There will be no splash screen included in the executable if following 
+% the instructions below.
+% In order to compile an executable with splash screen call deploytool and
+% add the required files manually.
+%
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%
+% Copyright 2015, Mark Bangert, on behalf of the matRad development team
+%
+% m.bangert@dkfz.de
+%
+% This file is part of matRad.
+%
+% matrad is free software: you can redistribute it and/or modify it under 
+% the terms of the GNU General Public License as published by the Free 
+% Software Foundation, either version 3 of the License, or (at your option)
+% any later version.
+%
+% matRad is distributed in the hope that it will be useful, but WITHOUT ANY
+% WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+% FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+% details.
+%
+% You should have received a copy of the GNU General Public License in the
+% file license.txt along with matRad. If not, see
+% <http://www.gnu.org/licenses/>.
+%
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+%% start the compiling process from the matRad root directory
+dirMatRad = pwd;                % set the matRad root directory
+dirOut = [pwd '\standalone\build\']; % specify the output folder
+
+if exist(dirOut)~= 7
+   mkdir(dirOut);
+end
+
+addpath('dicomImport');         % include the DICOM Import files in the dependency analysis
+
+
+mcc('-C','-o','matRad','-W','WinMain:matRad','-T','link:exe','-d',dirOut,...      
+    '-R','-logfile,consoleOutput.log','-v',[dirMatRad '\matRadGUI.m'],...
+    '-a',[dirMatRad '\dicomImport\DKFZ_Logo.png'],'-a',[dirMatRad '\dicomImport\matrad_logo.png'],...
+    '-a',[dirMatRad '\dicomImport\hlutDefault.mat'])
```

Comment:
The problem occurs not during the dose calculation but during the openingfcn, where the entries of the machine popup list are created (matRadGUI lines 103-120). dir(pattern) in line 105 will simply return an empty struct if the machine data is only included during the compilation process, therefore the application will crash.

In case of hlutDefault.mat the load command is used, which apparently does consider also the files added in the compilation (in opposite to dir). The same procedure worked also for the "old" photon/proton and carbon kernel files that where loaded without a selection, but as we now first search for available files this is not working anymore. I hope this was better understandable, but we can also discuss this issue in the next days if it is still unclear or if I missed a point!